aws_lambda_powertools/utilities/parser/models/kinesis.py

```diff
@@ -0,0 +1,37 @@
+import base64
+from binascii import Error as BinAsciiError
+from typing import List
+
+from pydantic import BaseModel, validator
+from pydantic.types import PositiveInt
+from typing_extensions import Literal
+
+
+class KinesisStreamRecordPayload(BaseModel):
+    kinesisSchemaVersion: str
+    partitionKey: str
+    sequenceNumber: PositiveInt
+    data: bytes  # base64 encoded str is parsed into bytes
+    approximateArrivalTimestamp: float
+
+    @validator("data", pre=True)
+    def data_base64_decode(cls, value):
+        try:
+            return base64.b64decode(value)
```

Comment:
Could you please initialize and add a logger here for debugging purposes?

```suggestion
            logger.debug("Decoding base64 Kinesis data record before parsing")
            return base64.b64decode(value)
```